xds/src/test/java/io/grpc/xds/LegacyLoadBalancerConfigFactoryTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+
+import com.google.protobuf.UInt32Value;
+import com.google.protobuf.UInt64Value;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster.LbPolicy;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster.LeastRequestLbConfig;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster.RingHashLbConfig;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster.RingHashLbConfig.HashFunction;
+import io.grpc.internal.JsonUtil;
+import io.grpc.internal.ServiceConfigUtil;
+import io.grpc.internal.ServiceConfigUtil.LbConfig;
+import io.grpc.xds.ClientXdsClient.ResourceInvalidException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Unit test for {@link LegacyLoadBalancerConfigFactory}.
+ */
+@RunWith(JUnit4.class)
+public class LegacyLoadBalancerConfigFactoryTest {
+
+  @Test
+  public void roundRobin() throws ResourceInvalidException {
+    Cluster cluster = Cluster.newBuilder().setLbPolicy(LbPolicy.ROUND_ROBIN).build();
+
+    LbConfig lbConfig = ServiceConfigUtil.unwrapLoadBalancingConfig(
+        LegacyLoadBalancerConfigFactory.newConfig(cluster, true));
+
+    assertThat(lbConfig.getPolicyName()).isEqualTo("round_robin");
+    assertThat(lbConfig.getRawConfigValue()).isEmpty();
+  }
+
+  @Test
+  public void ringHash() throws ResourceInvalidException {
+    Cluster cluster = Cluster.newBuilder().setLbPolicy(LbPolicy.RING_HASH).setRingHashLbConfig(
+        RingHashLbConfig.newBuilder()
+            .setMinimumRingSize(UInt64Value.newBuilder().setValue(1).build())
+            .setMaximumRingSize(UInt64Value.newBuilder().setValue(2).build()).build()).build();
+
+    LbConfig lbConfig = ServiceConfigUtil.unwrapLoadBalancingConfig(
+        LegacyLoadBalancerConfigFactory.newConfig(cluster, true));
+
+    assertThat(lbConfig.getPolicyName()).isEqualTo("ring_hash_experimental");
+    assertThat(JsonUtil.getNumberAsLong(lbConfig.getRawConfigValue(), "minRingSize")).isEqualTo(1);
```

Comment:
We might consider making a golden config and just assert that the produced config equals the golden config. I think the equals() would work out properly. This isn't bad, though.